hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonVectorizedRecordReader.java

```diff
@@ -68,6 +67,10 @@
   private AbstractDetailQueryResultIterator iterator;
 
   private QueryModel queryModel;
+  //This holds mapping of  fetch index with respect to project col index.
+  // it is used when same col is used in projection many times.So need to fetch only that col.
+  private List<Integer> projectionMapping = new ArrayList<>();
```

Comment:
Better use  an array to have fast lookups instead of list

Reply:
During initBatch only it is done(not frequent lookup/creation ).